packages/core/database/lib/entity-manager.js

```diff
@@ -398,10 +411,37 @@ const createEntityManager = (db) => {
             ...(data.__pivot || {}),
           }));
 
-          if (_.isEmpty(rows)) {
+          if (isEmpty(rows)) {
             continue;
           }
 
+          // delete previous relations
+          const typeAndFieldIdsMap = {};
+
+          for (const { field, related_type: relatedType, related_id: relatedId } of rows) {
+            if (!relatedType) continue; // if !relatedType then it's a component. No need to handle that case
+
+            const isMorphOne =
+              db.metadata.get(relatedType).attributes[field].relation === 'morphOne';
+            if (isMorphOne) {
+              const key = `${relatedType}+${field}`;
+              typeAndFieldIdsMap[key] = typeAndFieldIdsMap[key] || [];
+              typeAndFieldIdsMap[key].push(relatedId);
+            }
+          }
+
+          const orWhere = _.map(typeAndFieldIdsMap, (ids, typeAndField) => {
+            const [type, field] = typeAndField.split('+');
```

Comment:
That concat string approach might not look like a problem because there isn't supposed to be a `+` in the string but it's brittle code. 

You could use 2 groupBy to make the code safer 

I made an example with lodash fp lib here https://replit.com/@alexandrebodin/map#index.js